jerry-core/ecma/builtin-objects/ecma-builtin-object.c

```diff
@@ -109,15 +108,25 @@ ecma_builtin_object_object_get_prototype_of (ecma_value_t this_arg, /**< 'this'
 {
   JERRY_UNUSED (this_arg);
   ecma_value_t ret_value = ECMA_VALUE_EMPTY;
+  bool was_object = ecma_is_value_object (arg);
 
   /* 1. */
-  if (!ecma_is_value_object (arg))
+  if (!was_object)
   {
+#ifndef CONFIG_DISABLE_ES2015_BUILTIN
+    ret_value = ecma_op_to_object (arg);
+    if (ECMA_IS_VALUE_ERROR (ret_value))
+    {
+      return ret_value;
+    }
+    arg = ret_value;
+#else /* CONFIG_DISABLE_ES2015_BUILTIN */
     ret_value = ecma_raise_type_error (ECMA_ERR_MSG ("Argument is not an object."));
```

Comment:
`return ecma_raise_type_error...`

I think this makes the `if (ecma_is_value_object (arg))` below unnecessary.